projects/saturn/src/main/java/io/fairspace/saturn/webdav/DavFactory.java

```diff
@@ -126,27 +128,33 @@ org.apache.jena.rdf.model.Resource currentUserResource() {
         return rootSubject.getModel().createResource(getUserURI().getURI());
     }
 
-    public boolean isFileSystemResource(org.apache.jena.rdf.model.Resource resource) {
-        return resource.getURI().startsWith(rootSubject.getURI());
-    }
-
-    public void linkEntityToSubject(org.apache.jena.rdf.model.Resource subject) throws BadRequestException {
-        var linkedEntity = getNewLinkedEntity(subject);
+    public void linkEntityToSubject(org.apache.jena.rdf.model.Resource subject) throws BadRequestException, NotAuthorizedException {
+        var linkedEntity = getLinkedEntity(subject);
         subject.addProperty(FS.linkedEntity, linkedEntity);
         subject.addProperty(FS.linkedEntityType, linkedEntity.getPropertyResourceValue(RDF.type));
     }
 
-    private org.apache.jena.rdf.model.Resource getNewLinkedEntity(org.apache.jena.rdf.model.Resource subject) throws BadRequestException {
+    public void validateAuthorization(String typeUri) throws NotAuthorizedException {
```

Comment:
Just a small remark, could this method be named differently, e.g. `validateCanWrite` or `validateCanCreateDirectory`? It now seems that it can be used for all the methods to authorize user actions, while it only checks for writing access, depending on the entity type.